test/unit/test_random.c

```diff
@@ -0,0 +1,54 @@
+#include "unittest.h"
+#include "../../src/random.h"
+
+static int setup(void **state) {
+	(void)state;
+	random_init();
+	return 0;
+}
+
+static int teardown(void **state) {
+	(void)state;
+	random_exit();
+	return 0;
+}
+
+static const uint8_t ref[128] = {0};
+
+static void test_randomize_zero_must_not_change_memory(void **state) {
+	(void)state;
+
+	uint8_t data[sizeof(ref)] = {0};
+	randomize(data, 0);
+
+	assert_memory_equal(ref, data, sizeof(data));
+}
+
+static void test_randomize_half_changes_half(void **state) {
+	(void)state;
+
+	uint8_t data[sizeof(ref)] = {0};
+	const size_t half = sizeof(data) / 2;
+	randomize(data, half);
+
+	assert_memory_not_equal(data, ref, half);
+	assert_memory_equal(&data[half], ref, half);
+}
+
+static void test_randomize_full_changes_memory(void **state) {
+	(void)state;
+
+	uint8_t data[sizeof(ref)] = {0};
+	randomize(data, sizeof(data));
+
+	assert_memory_not_equal(ref, data, sizeof(data));
+}
+
+int main(void) {
+	const struct CMUnitTest tests[] = {
+		cmocka_unit_test(test_randomize_zero_must_not_change_memory),
+		cmocka_unit_test(test_randomize_half_changes_half),
+		cmocka_unit_test(test_randomize_full_changes_memory),
```

Comment:
I think it might then also be nice that `randomize()` produces different data every time it is called.

Reply:
I thought about adding statistical tests, but it's a bit too much. This is good middle ground.

Also added a check that `randomize()` fails if you forget to initialize it.

The best way to do that would probably be [mocking `assert`][assert], but it requires rebuilding the project with a different set of compiler flags.

[assert]: https://api.cmocka.org/group__cmocka__mock__assert.html

Current meson configs don't allow that, so it's a relatively big task. I'll look into that separately.

I tried a bunch of other hacks (handling `SIGABRT`, mocking `abort()`, mocking a wrapper around `abort()`, `longjmp` trickery, `fork` + `wait`). Those turned out to be ugly for one reason or another.

If you'd rather wait for a proper solution, I'll remove the test for now.